### PR TITLE
More repository/sql changes.

### PR DESCRIPTION
**Pull request by agorina**

### CODE REVIEW

File: reporting-service/src/test/java/org/opentestsystem/rdw/reporting/exam/group/JdbcGroupAssessmentRepositoryIT.java [hunk 1]
```diff
@@ -1,92 +1,67 @@
...
-    @After
-    public void clearState() {
-        tenantProperties.setTransferAccessEnabled(false);
-    }
+    private GroupAssessmentRepository repository;
 
     private GroupAssessmentSearch.Builder search() {
         return GroupAssessmentSearch.builder()
-                .groupGrant(new GroupGrant(-10L, 1))
+                .groupGrant(new GroupGrant(-10L, 99999)) //TODO: the last parameter is outdated and needs to be removed
```
Comment (agorina) [inline reporting-service/src/test/java/org/opentestsystem/rdw/reporting/exam/group/JdbcGroupAssessmentRepositoryIT.java]: I will create a JIRA to clean up GroupGrant after this is done. We will not be used a group subject, instead we look it up by the given group id.

File: reporting-service/src/main/java/org/opentestsystem/rdw/reporting/exam/group/JdbcGroupAssessmentRepository.java [hunk 2]
```diff
@@ -17,16 +17,15 @@
     JdbcGroupAssessmentRepository(
             final NamedParameterJdbcTemplate template,
             final SecurityParameterProvider securityParameterProvider,
-            @Value("${sql.group.exam.findLatest}") final String findLatestQuery,
+            @Value("${sql.group.assessment.findLatestExam}") final String findLatestQuery,
```
Comment (agorina) [inline reporting-service/src/main/java/org/opentestsystem/rdw/reporting/exam/group/JdbcGroupAssessmentRepository.java]: Renamed to match the repository name.

File: reporting-service/src/main/java/org/opentestsystem/rdw/reporting/exam/group/JdbcGroupAssessmentRepository.java [hunk 3]
```diff
@@ -17,16 +17,15 @@
...
             @Value("${sql.group.assessment.findAll}") final String findAllQuery) {
         super(template, securityParameterProvider, findLatestQuery, findAllQuery);
     }
 
     @Override
     protected Map<String, Object> getParameters(final GroupAssessmentSearch search) {
         return ImmutableMap.of(
-                "group_id", search.getGroupGrant().getId(),
-                "group_subject_id", search.getGroupGrant().getSubjectId()
```
Comment (agorina) [inline reporting-service/src/main/java/org/opentestsystem/rdw/reporting/exam/group/JdbcGroupAssessmentRepository.java]: We do not pass the subject id anymore. It is being derived from the group.

File: reporting-service/src/test/java/org/opentestsystem/rdw/reporting/exam/group/JdbcGroupAssessmentRepositoryIT.java [hunk 4]
```diff
@@ -1,92 +1,67 @@
...
     @Test
     public void findAllShouldFindAll() throws Exception {
-        assertThat(repository.findAll(PermissionScope.STATEWIDE, search().build()))
+        assertThat(repository.findAll(userBuilder.permissionsById(groupStatewide).build(), search().build()))
                 .usingRecursiveFieldByFieldElementComparator()
                 .containsOnlyElementsOf(TestData.STUDENT10_GROUP10_ASSESSMENTS_WITH_CUT_POINTS);
     }
 
+    @Test
+    public void findAllShouldFindAllForGroupWithAllSubjects() throws Exception {
```
Comment (agorina) [inline reporting-service/src/test/java/org/opentestsystem/rdw/reporting/exam/group/JdbcGroupAssessmentRepositoryIT.java]: Added missing test.

File: reporting-service/src/test/java/org/opentestsystem/rdw/reporting/exam/schoolgrade/JdbcSchoolGradeAssessmentRepositoryIT.java [hunk 5]
```diff
@@ -77,13 +58,11 @@ public void itShouldFindAllAssessmentsRespectingTransferAccess() throws Exceptio
                 .schoolId(-40)
                 .schoolYear(1998)
                 .build();
-        final List<Assessment> withoutTransfer = repository.findAll(PermissionScope.STATEWIDE, query);
-        assertThat(withoutTransfer.stream().map(Assessment::getId))
-                .containsOnly(-2L);
 
-        tenantProperties.setTransferAccessEnabled(true);
-        final List<Assessment> withTransfer = repository.findAll(PermissionScope.STATEWIDE, query);
```
Comment (agorina) [inline reporting-service/src/test/java/org/opentestsystem/rdw/reporting/exam/schoolgrade/JdbcSchoolGradeAssessmentRepositoryIT.java]: I noticed that I changed this test but I wonder if it was right. With `PermissionScope.STATEWIDE`, how could there be a diff. with the `setTransferAccessEnabled`?

Am I missing something?

File: reporting-service/src/main/resources/application.sql.yml [hunk 6]
```diff
@@ -108,26 +107,27 @@ sql:
               join school isch on st.inferred_school_id = isch.id
             where e.school_year=:school_year
             and a.id = e.asmt_id
-            and ${sql.reporting.snippet.schoolScopePermission}
+            and isch.id = :school_id
```
Comment (agorina) [inline reporting-service/src/main/resources/application.sql.yml]: This does not seem to make any difference for the result. I am afraid I am missing something here. Please see my notes below about the test that is using it.

File: reporting-service/src/main/resources/application.sql.yml [hunk 7]
```diff
@@ -108,26 +107,27 @@ sql:
               join school isch on st.inferred_school_id = isch.id
             where e.school_year=:school_year
             and a.id = e.asmt_id
-            and ${sql.reporting.snippet.schoolScopePermission}
+            and (s.id = :school_id OR (1 = :allow_transfer_access && isch.id = :school_id))
```
Comment (agorina) [inline reporting-service/src/main/resources/application.sql.yml]: After a few attempts, I think I figured what we need here, but would like a confirmation.

This returns all assessments for the given school/grade/year. 
If a `transfer access` is enabled, we also want to return assessments for the students that are currently enrolled into the school but may have taken the given grade assessment  outside of the given school.

Comment (wtritch) [reply reporting-service/src/main/resources/application.sql.yml]: Yep, that’s pretty much it.  “Transfer Access” isn’t just about permissions it’s also changing the result set to include results from institutions that a “student in your selected institution” took previously.

For example if you want all 2017 6th grade ELAs for George Washington middle school and Bobby Student just transferred in in Jan from Another Middle School, you’d include Bobby’s Another Middle School results in the result set.

Comment (agorina) [reply reporting-service/src/main/resources/application.sql.yml]: Makes sense, thank you!

**Pull request merged**